### PR TITLE
perf: disable kvcache for semantic by default

### PR DESCRIPTION
**Pull request by no2chem**
An analysis of my previous patches suggests that use_kv_caching=True is not a win at all for the text model.

The current KV caching code has quite a bit of overhead. In particular, it (re)allocates large tensors with adding to the cache via a `torch.cat` call. With the coarse model it seems to result in some performance gain, but it is definitely not a win for the smaller text model. It seems that the KV cache was what was causing the bimodality in #366, I'm guessing sometimes we would get unlucky and the model would have to reallocate.

With this change I can consistently get about 280 it/second performance for the text model on an H100 after warmup.

### CODE REVIEW

**Comment by no2chem** [comment]
Hm, nevermind here, I think I drew an invalid conclusion due to a mixup. It seems that caching still results in a small performance gain on the text model.
